### PR TITLE
Generate API docs for EKS

### PR DESCRIPTION
**Pull request by lblackstone**
Fix https://github.com/pulumi/pulumi-eks/issues/447

### CODE REVIEW

**Comment by lblackstone** [comment]
@justinvp This should be RFR. Here's the direct link: http://pulumi-docs-origin-pr-4696-e0c75b8c.s3-website.us-west-2.amazonaws.com/docs/reference/pkg/eks/

File: content/docs/reference/pkg/eks/_index.md
Comment (justinvp) [inline content/docs/reference/pkg/eks/_index.md:22]: In EKS, I'm currently deleting the generated `Provider` resource as I didn't think it'd be useful for this library. But forgot that it would show up here in the docs. I suppose I'll just go ahead and bring this back in EKS.

**Review by justinvp — APPROVED** [review]
LGTM

I guess we'll also want to add it https://github.com/pulumi/docs/blob/master/scripts/gen_all_resource_docs.sh, but looks like we need to ensure the Makefile has a `generate_schema` target.

**Comment by lblackstone** [comment]
> I guess we'll also want to add it https://github.com/pulumi/docs/blob/master/scripts/gen_all_resource_docs.sh, but looks like we need to ensure the Makefile has a `generate_schema` target.

https://github.com/pulumi/pulumi-eks/pull/518

**Comment by lblackstone** [comment]
Regenerated with the real v0.22 tag, and will merge once CI is ✅

**Comment by lblackstone** [comment]
@davidwrede Once this merges, it would be good to update the [index page](https://www.pulumi.com/docs/reference/pkg/) to point to the new EKS docs. It shows up in the left sidebar index, but there's also an outdated reference near the bottom under the TypeScript/JavaScript chooser that links to the old npm docs.

**Comment by davidwrede** [comment]
Thanks @lblackstone. I've started some initial thinking on this in https://github.com/pulumi/docs/issues/4966.

**Pull request merged**
